src/NUnitFramework/tests/SynchronizationContextTests.cs

```diff
@@ -104,7 +104,7 @@ public static void AsyncDelegatesAreExecutedOnStaThread(AsyncExecutionApiAdapter
             });
         }
 
-#if NETFRAMEWORK
+#if !NETCOREAPP2_1
```

Comment:
This change is what is causing the CI build failure. `System.Windows.Threading.DispatcherSynchronizationContext` and `System.Windows.Threading.DispatcherSynchronizationContext` aren't available to compile or run against in .NET Core 3.1 unless we use the Windows desktop SDK (which fails the build on other OSs) or come up with a workaround.